llvm/lib/Target/AMDGPU/GCNDPPCombine.cpp

```diff
@@ -781,3 +792,14 @@ bool GCNDPPCombine::runOnMachineFunction(MachineFunction &MF) {
   }
   return Changed;
 }
+
+PreservedAnalyses GCNDPPCombinePass::run(MachineFunction &MF,
+                                         MachineFunctionAnalysisManager &) {
```

Comment:
Add the following to skip this pass for OptNone cases.
if (MF.getFunction().hasOptNone())
   return false;